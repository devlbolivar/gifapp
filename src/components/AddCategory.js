import React,{useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {

        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories(categories => [ inputValue,...categories]);
            setInputValue('');
        }

    }


    return (
        <form className="form__search__bar" onSubmit={handleSubmit}>
            <input
                type="text"
                value={inputValue}
                placeholder="Add category"
                onChange={handleInputChange}
            />
            <button>ok</button>
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}