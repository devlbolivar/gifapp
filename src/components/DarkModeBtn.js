import React from 'react'

export const DarkModeBtn = () => {
    return (
        <div className="push">
            <span>bot√≥n</span>
        </div>
    );
}
