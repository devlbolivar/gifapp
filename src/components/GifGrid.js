//import React, { useState, useEffect } from 'react'
import React from 'react'
import { GifGridItem } from './GifGridItem';
import {useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs(category);
    
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">loading</p> }
            <div className="card-grid">
                { 
                    images.map(image =>(
                        <GifGridItem 
                            key={image.id} 
                            {...image} 
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
