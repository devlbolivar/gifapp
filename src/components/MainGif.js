import React from "react";
import { useFetchGif } from "../hooks/useFetchGif";

export const MainGif = () => {
  useFetchGif();
  return (
    <div className="main__gif">
      <img alt="Random Img logo" />
    </div>
  );
};
