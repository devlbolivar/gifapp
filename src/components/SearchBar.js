import React from 'react';
import { AddCategory } from './AddCategory';

export const SearchBar = ({setCategories}) => {
    return (
        <div>
            <AddCategory setCategories={setCategories} />
        </div>
    )
}
