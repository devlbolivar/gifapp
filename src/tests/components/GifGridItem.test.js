import React from 'react';
import { shallow } from 'enzyme';
import {GifGridItem} from '../../components/GifGridItem';

describe('Prueba inicial del componente <GifGridItem />', () => {
    
    const title = 'Título de prueba';
    const url = 'https://urldeprueba.com';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('Debe mostrar <GifGridItem /> correctamente', () =>{
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un párrafo con  el title', () =>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('Debe de tener la img igual al url y alt de los props', () =>{
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe de tener animate__fadeIn', () =>{
        const div = wrapper.find('div');
        //console.log(div.props());
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
    });
});