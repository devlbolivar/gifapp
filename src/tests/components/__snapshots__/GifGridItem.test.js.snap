// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba inicial del componente <GifGridItem /> Debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Título de prueba"
    src="https://urldeprueba.com"
  />
  <p>
    Título de prueba
  </p>
</div>
`;
